### ISSUE TITLE
Weird state of checkTrialOrIntroDiscountEligibility

### ISSUE THREAD

**Issue by Pattio** [RevenueCat/purchases-ios#3862]
- [ ] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-ios/blob/main/Contributing/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-ios/issues)
------------

There's an interesting issue regarding `checkTrialOrIntroDiscountEligibility`:

It doesn't seem to work as it consistently returns `unknown` This [issue was previously reported, and the solution provided was to use StoreKit 2](https://github.com/RevenueCat/purchases-ios/issues/1746). However, [a later ticket suggests using StoreKit 1 instead](https://github.com/RevenueCat/purchases-ios/issues/1893). Additionally, if we use `useStoreKit2IfAvailable`, we receive a warning that it's deprecated and that we shouldn't set a value on it.

This leads to a confusing cycle:

- StoreKit 2 is slow; use StoreKit 1.
- StoreKit 1 doesn't work; use StoreKit 2.
- `useStoreKit2IfAvailable` is deprecated; you shouldn't set this value.

In the end, I'm left confused. In my experience, StoreKit 1 doesn't work, and while StoreKit 2 seems to function, other warnings suggest that we shouldn't use it.

-----------

1. Environment
   1. Platform: macOS
   4. SDK version: `4.41.0`
   5. StoreKit version:
      - [x] StoreKit 1
      - [x] StoreKit 2 (_enabled with `usesStoreKit2IfAvailable(true)`_)
   6. OS version: 14.4.1 (23E224)
   7. Xcode version: 15.3.0
   8. Device and/or simulator:
      - [x] Device
      - [ ] Simulator
   9. Environment:
      - [] Sandbox
      - [ ] TestFight
      - [ ] Production
   10. How widespread is the issue. Percentage of devices affected.

**Comment by RCGitBot** [comment]
👀 We've just linked this [issue to our internal tracker](https://revenuecat.zendesk.com/agent/tickets/40312) and notified the team. Thank you for reporting, we're checking this out!

**Comment by Pattio** [comment]
The reason why I think it's pretty major is the fact that afaik without `checkTrialOrIntroDiscountEligibility` you can't really properly offer trials and intro discounts which is one of the bigger selling points when using RevenueCat

**Comment by WesleyClarkRC** [comment]
Hi! We would recommend utilizing StoreKit2 with `useStoreKit2IfAvailable`. The call is deprecated, but not in functionality and it will continue to work. We are working towards a new call site for it. Can you share the debug logs for this when you are getting this unknown response?

**Comment by Pattio** [comment]
Hey, Wesley, thanks for the response!

> We would recommend utilizing StoreKit2 with useStoreKit2IfAvailable.

Is this an official recommendation? I couldn't find any documentation or anything on your site that would suggest this.

I had thought of using `useStoreKit2IfAvailable`  but then I was hesitant, because:
1. I saw quite a few issues on the iOS repo with the RevenueCat team suggesting staying with SK1 as it's more stable/safe, for example [this](https://github.com/RevenueCat/purchases-ios/issues/2251#issuecomment-1414369765)
2. Going off the default path (since SK1 is the default when you init SDK) seems more dangerous 

> Can you share the debug logs for this when you are getting this unknown response?

Yes. I found the error which is:
> WARN: 🍎‼️ App running on sandbox without a receipt file. Unable to determine intro eligibility unless you've purchased before and there is a receipt available.

I still haven't uploaded my app to the Mac App Store, so if this error will go away for people who download the app from App Store (as noted [here](https://github.com/RevenueCat/purchases-ios/issues/2580#issuecomment-1581212942)) then I'm fine keeping SK1
